bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1368,8 +1368,16 @@ def test_checksum_match_immediately(self, urlopen, publish):
             mock.call(topic='mashtask.sync.done', msg={'repo': t.id, 'agent': 'bowlofeggs'},
                       force=True)]
         publish.assert_has_calls(expected_calls)
-        urlopen.assert_called_once_with(
-            'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
+        self.assertEqual(urlopen.call_count, 1)
+        # Since os.listdir() isn't deterministic about the order of the items it returns, the test
+        # won't be deterministic about which of these URLs get called. However, either one of them
+        # would be correct so we will just assert that one of them is called.
```

Comment:
@pypingou I would, but actually only one of them gets used and I can't predict which one that would be without further mocking. So I decided to just make sure that the one that gets used is one of the two because it doesn't really matter which one gets used. It's weird ☺